Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -91,24 +99,37 @@ void GridMasonryLayout::addItemsToFirstTrack(const HashMap<RenderBox*, GridArea>
         insertIntoGridAndLayoutItem(item, gridArea);
 }
 
+void GridMasonryLayout::placeItemsUsingOrderModifiedDocumentOrder()
+{
+    for (auto* child = m_renderGrid.currentGrid().orderIterator().first(); child; child = m_renderGrid.currentGrid().orderIterator().next()) {
+        if (m_renderGrid.currentGrid().orderIterator().shouldSkipChild(*child) || m_firstTrackItems.contains(child))
```

Comment:
Is it potentially a performance problem that we have to do a hash map lookup into the first track items, for every child?

Could instead `collectMasonryItems` just put all non-first track items into one of the two definite/indefinite lists (doesn't matter which, just unconditionally put them all in one if definite-first is enabled).

We could then iterate that list in this function, knowing that it already excludes the first track children (and is in the right order).

Reply:
Yeah I could see that becoming an issue. Changed the logic to store all of the items in the definite items vector.